builtin/update-index.c

```diff
@@ -1197,7 +1233,6 @@ int cmd_update_index(int argc, const char **argv, const char *prefix)
 		remove_split_index(&the_index);
 	}
 
-	prepare_repo_settings(r);
```

Comment:
This extra call will not be harmful, but removing it might cause a bug in the future (if/when we remove `command_requires_full_index` we might also remove the other `prepare_repo_settings()` and cause a bug). The method does nothing if the settings are already prepared, so having an extra call is not a performance problem. It's a good pattern to call `prepare_repo_settings()` immediately before using one of its settings.

Reply:
Good to know! Added back in.